### PR TITLE
Update parameter names for better readability

### PR DESCRIPTION
**Pull request by chatton**
## Description

closes: #5463

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

File: modules/apps/transfer/ibc_module.go [hunk 1]
```diff
@@ -344,7 +344,7 @@ func (IBCModule) OnChanUpgradeAck(ctx sdk.Context, portID, channelID, counterpar
 }
 
 // OnChanUpgradeOpen implements the IBCModule interface
-func (IBCModule) OnChanUpgradeOpen(ctx sdk.Context, portID, channelID string, order channeltypes.Order, connectionHops []string, version string) {
+func (IBCModule) OnChanUpgradeOpen(ctx sdk.Context, portID, channelID string, proposedOrder channeltypes.Order, proposedConnectionHops []string, proposedVersion string) {
```
Comment (crodriguezvega) [inline modules/apps/transfer/ibc_module.go:347]:
```suggestion
func (IBCModule) OnChanUpgradeOpen(ctx sdk.Context, portID, channelID string, proposedOrder channeltypes.Order, proposedConnectionHops []string, counterpartyVersion string) {
```

File: modules/apps/callbacks/ibc_middleware.go [hunk 2]
```diff
@@ -365,23 +365,23 @@ func (im IBCMiddleware) OnChanCloseConfirm(ctx sdk.Context, portID, channelID st
...
-func (im IBCMiddleware) OnChanUpgradeTry(ctx sdk.Context, portID, channelID string, order channeltypes.Order, connectionHops []string, counterpartyVersion string) (string, error) {
+func (im IBCMiddleware) OnChanUpgradeTry(ctx sdk.Context, portID, channelID string, proposedOrder channeltypes.Order, proposedConnectionHops []string, counterpartyVersion string) (string, error) {
 	cbs, ok := im.app.(porttypes.UpgradableModule)
 	if !ok {
 		return "", errorsmod.Wrap(porttypes.ErrInvalidRoute, "upgrade route not found to module in application callstack")
 	}
 
-	return cbs.OnChanUpgradeTry(ctx, portID, channelID, order, connectionHops, counterpartyVersion)
+	return cbs.OnChanUpgradeTry(ctx, portID, channelID, proposedOrder, proposedConnectionHops, counterpartyVersion)
 }
```
Comment (crodriguezvega) [inline modules/apps/callbacks/ibc_middleware.go:385]: Should we rename the parameters in the `OnChanUpgradeOpen` callback as well?

File: modules/apps/27-interchain-accounts/controller/ibc_middleware.go [hunk 3]
```diff
@@ -293,7 +293,7 @@ func (im IBCMiddleware) OnChanUpgradeAck(ctx sdk.Context, portID, channelID, cou
 }
 
 // OnChanUpgradeOpen implements the IBCModule interface
-func (im IBCMiddleware) OnChanUpgradeOpen(ctx sdk.Context, portID, channelID string, order channeltypes.Order, connectionHops []string, version string) {
+func (im IBCMiddleware) OnChanUpgradeOpen(ctx sdk.Context, portID, channelID string, proposedOrder channeltypes.Order, proposedConnectionHops []string, proposedVersion string) {
```
Comment (crodriguezvega) [inline modules/apps/27-interchain-accounts/controller/ibc_middleware.go:296]:
```suggestion
func (im IBCMiddleware) OnChanUpgradeOpen(ctx sdk.Context, portID, channelID string, proposedOrder channeltypes.Order, proposedConnectionHops []string, counterpartyVersion string) {
```

**Review by crodriguezvega — COMMENTED** [review]
I think there a couple of places where we could do some extra renaming.

File: modules/apps/27-interchain-accounts/controller/ibc_middleware.go [hunk 4]
```diff
@@ -254,14 +254,14 @@ func (im IBCMiddleware) OnChanUpgradeInit(ctx sdk.Context, portID, channelID str
...
+		return cbs.OnChanUpgradeInit(ctx, portID, channelID, proposedOrder, proposedConnectionHops, proposedVersion)
 	}
 
-	return version, nil
+	return proposedVersion, nil
 }
 
 // OnChanUpgradeTry implements the IBCModule interface
-func (IBCMiddleware) OnChanUpgradeTry(ctx sdk.Context, portID, channelID string, order channeltypes.Order, connectionHops []string, counterpartyVersion string) (string, error) {
+func (IBCMiddleware) OnChanUpgradeTry(_ sdk.Context, portID, channelID string, proposedOrder channeltypes.Order, proposedConnectionHops []string, counterpartyVersion string) (string, error) {
```
Comment (damiannolan) [inline modules/apps/27-interchain-accounts/controller/ibc_middleware.go:264]: Just wondering if we want to discard (`_`)  unused args or leave them in place - I don't know what the linters stance on this is.

File: modules/core/05-port/types/module.go [hunk 5]
```diff
@@ -139,9 +139,9 @@ type UpgradableModule interface {
 		ctx sdk.Context,
 		portID,
 		channelID string,
-		order channeltypes.Order,
-		connectionHops []string,
-		version string,
+		proposedOrder channeltypes.Order,
+		proposedConnectionHops []string,
+		counterpartyVersion string,
```
Comment (colin-axner) [inline modules/core/05-port/types/module.go:144]: this should be proposed version? It is pulled from the upgrade.fields.version

File: modules/apps/27-interchain-accounts/controller/ibc_middleware.go (hunk 4 above)
Comment (chatton) [reply modules/apps/27-interchain-accounts/controller/ibc_middleware.go:264]: we could alright, I don't feel too strongly either way

Comment (DimitrisJim) [reply modules/apps/27-interchain-accounts/controller/ibc_middleware.go:264]: pretty sure we removed linter that couldn't cope with this.

**Review by colin-axner — APPROVED** [review]
ACK but see suggestions, some consistency is still missing. Thank you!

**Review by crodriguezvega — APPROVED** [review]

**Pull request merged**